[2.9.0] Feature Flag - external-rules (#1361)
rancher/rancher-docs: docs/getting-started/installation-and-upgrade/installation-references/feature-flags.md

The following is a list of feature flags available in Rancher. If you've upgraded from a previous Rancher version, you may see additional flags in the Rancher UI, such as `proxy` or `dashboard` (both [discontinued](/versioned_docs/version-2.5/reference-guides/installation-references/feature-flags.md)):

- `continuous-delivery`: Allows Fleet GitOps to be disabled separately from Fleet. See [Continuous Delivery.](../../../how-to-guides/advanced-user-guides/enable-experimental-features/continuous-delivery.md) for more information.

> - `external-rules`: This flag affected [external `RoleTemplate` behavior](../../../how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/manage-role-based-access-control-rbac/cluster-and-project-roles.md#external-roletemplate-behavior) and is removed in v2.9.0 and later Rancher versions as the flag remains on by default.

Reviewer: ```suggestion
- `external-rules`: This flag affected [external `RoleTemplate` behavior](../../../how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/manage-role-based-access-control-rbac/cluster-and-project-roles.md#external-roletemplate-behavior). It is removed in v2.9.0 and later as the behavior is enabled by default.
```

> - `external-rules`: This flag affected [external `RoleTemplate` behavior](../../../how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/manage-role-based-access-control-rbac/cluster-and-project-roles.md#external-roletemplate-behavior) and is removed in v2.9.0 and later Rancher versions as the flag remains on by default.

Reviewer: ```suggestion
`external-rules`: This flag affected [external `RoleTemplate` behavior](../../../how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/manage-role-based-access-control-rbac/cluster-and-project-roles.md#external-roletemplate-behavior). It is removed in Rancher v2.9.0 and later as the behavior is enabled by default.
```

Author: Applied suggestion.

> - `external-rules`: This flag is removed in this current and later Rancher versions as the flag always remains on. With its default enablement, external `RoleTemplates` can be created only if the backing `ClusterRole` exists in the local cluster or the `ExternalRules` is set. Note that `escalate` permissions on `RoleTemplates` are required to create external `RoleTemplates` with `ExternalRules`. For context, the backing `ClusterRole` holds cluster rules and privileges, and shares the same `metadata.name` used in the `RoleTemplate` in your respective cluster referenced by the `ClusterRoleTemplateBinding/ProjectRoleTemplateBinding`.

Reviewer: ```suggestion
- `external-rules`: In Rancher v2.9.0 and later, this flag is removed from the UI as the setting is permanently enabled. In previous Rancher versions, it is on by default. With this setting enabled, external `RoleTemplates` can be created only if the backing `ClusterRole` exists in the local cluster or the `ExternalRules` is set. Note that `escalate` permissions on `RoleTemplates` are required to create external `RoleTemplates` with `ExternalRules`. For context, the backing `ClusterRole` holds cluster rules and privileges, and shares the same `metadata.name` used in the `RoleTemplate` in your respective cluster referenced by the `ClusterRoleTemplateBinding/ProjectRoleTemplateBinding`.
```

The following table shows the availability and default values for some feature flags in Rancher. Features marked "GA" are generally available:

| Feature Flag Name             | Default Value | Status       | Available As Of | Additional Information |
| ----------------------------- | ------------- | ------------ | --------------- | ---------------------- |
| `continuous-delivery` | `true` | GA | v2.6.0 | |
| `external-rules` | v2.7.14: `false`, v2.8.5: `true` | Removed | v2.7.14, v2.8.5 | This flag affected [external `RoleTemplate` behavior](../../../how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/manage-role-based-access-control-rbac/cluster-and-project-roles.md#external-roletemplate-behavior). It is removed in Rancher v2.9.0 and later as the behavior is enabled by default. |
| `fleet`  | `true` | Can no longer be disabled | v2.6.0 | |
| `fleet`  | `true` | GA | v2.5.0 | |
| `harvester` | `true` | Experimental | v2.6.1 | |
| `legacy` | `false` for new installs, `true` for upgrades | GA | v2.6.0 | |
| `rke1-custom-node-cleanup`| `true` | GA | v2.6.0 | |
| `rke2` | `true` | Experimental | v2.6.0 | |

> | `token-hashing` | `false` for new installs, `true` for upgrades | GA | v2.6.0 | |

Reviewer 2: ```suggestion
| `token-hashing` | `false` for new installs, `true` for upgrades | GA | v2.6.0 | |
| `ui-sql-cache` | `false` | Highly experimental | v2.9.0 |
```

Line 47 from this diff shouldn't be removed since it was just merged as part of https://github.com/rancher/rancher-docs/pull/1337.
